app/views/pages/signup.ejs

```diff
@@ -0,0 +1,33 @@
+<%- include('../partials/header'); %> 
+<%- include('../partials/navbar'); %>
+
+<main class="container-xl">
+    <% if (err) { %>
+        <div class="alert alert-danger" role="alert"><%= err %></div>
+    <% } %>  
+    <h1>Sign Up</h1>
+    <form action="/api/signup" method="post">
+        <div class="mb-3">
+            <label for="username" class="form-label">Email</label>
+            <input id="email" name="email" type="email" class="form-control" autocomplete="email" required>
+        </div>
+        <div class="mb-3">
+            <label for="new-password" class="form-label">Password</label>
+            <input id="new-password" name="password" type="password" class="form-control" autocomplete="new-password" required name="password" 
+            pattern="^(?=.*\d)(?=.*[a-zA-Z])(?!.*\s).{8,100}$" 
+            title="Password must be between 8 and 100 characters long, contain no spaces, and include at least one digit and one letter.">
+        </div>
```

Comment:
Like we discussed in-call: we are adding a confirm password field, then I will approve/merge once that is finished 👍

Reply:
Added the confirmation field and logic.